classes/class-wc-connect-label-reports.php

```diff
@@ -0,0 +1,221 @@
+				class="export_csv"
+				data-export="table"
+			>
+				<?php _e( 'Export CSV', 'woocommerce-services' ); ?>
+			</a>
+			<?php
+		}
+
+		private function compare_label_dates_desc( $label_a, $label_b ) {
+			return $label_b['created'] - $label_a['created'];
+		}
+
+		private function get_all_labels() {
+			global $wpdb;
+			$query = "SELECT post_id, meta_value FROM {$wpdb->postmeta} WHERE meta_key = 'wc_connect_labels'";
+			$db_results = $wpdb->get_results( $query );
+			$results = array();
+
+			foreach ( $db_results as $meta ) {
+				$labels = maybe_unserialize( $meta->meta_value );
+
+				if ( ! is_array( $labels ) ) {
+					$labels = $this->settings_store->try_deserialize_labels_json( $meta->meta_value );
+				}
+
+				if ( empty( $labels ) ) {
+					continue;
+				}
+
+				foreach ( $labels as $label ) {
+					$results[] = array_merge( $label, array( 'order_id' => $meta->post_id ) );
+				}
+			}
+
+			usort( $results, array( $this, 'compare_label_dates_desc' ) );
+			return $results;
+		}
+
+		private function query_labels() {
+			$all_labels =  wp_cache_get( self::LABELS_CACHE_KEY );
+			if ( false === $all_labels ) {
+				$all_labels = $this->get_all_labels();
+				wp_cache_set( self::LABELS_CACHE_KEY, $all_labels, '', 1800 );
+			}
+
+			// translate timestamps to JS timestapms
+			$start_date = $this->start_date * 1000;
+			$end_date = $this->end_date * 1000;
+
+			$results = array();
+			foreach ( $all_labels as $label ) {
+				$created = $label['created'];
+				if ( $created > $end_date ) {
+					continue;
+				}
+
+				//labels are sorted in descending order, so if we reached the end, break the loop
+				if ( $created < $start_date ) {
+					break;
+				}
+
+				if ( isset( $label['error'] ) || ! isset( $label['rate'] ) ) {
+					continue;
+				}
+
+				$results[] = $label;
+			}
+
+			return $results;
+		}
+
+		public function output_report() {
+			$ranges = array(
+				'year'       => __( 'Year', 'woocommerce-services' ),
+				'last_month' => __( 'Last month', 'woocommerce-services' ),
+				'month'      => __( 'This month', 'woocommerce-services' ),
+				'7day'       => __( 'Last 7 days', 'woocommerce-services' ),
+			);
+
+			$current_range = ! empty( $_GET['range'] ) ? sanitize_text_field( $_GET['range'] ) : '7day';
+
+			if ( ! in_array( $current_range, array( 'custom', 'year', 'last_month', 'month', '7day' ) ) ) {
+				$current_range = '7day';
+			}
+
+			$this->check_current_range_nonce( $current_range );
+			$this->calculate_current_range( $current_range );
+
+			$hide_sidebar = true;
+
+			include( WC()->plugin_path() . '/includes/admin/views/html-report-by-date.php' );
+		}
+
+		private function get_order_url( $post_id ) {
+			$order = wc_get_order( $post_id );
+			return '<a href="' . WC_Connect_Compatibility::instance()->get_edit_order_url( $order ) . '">' . WC_Connect_Compatibility::instance()->get_order_id( $order ) . '</a>';
```

Comment:
This should be retrieving the order number, not the order id. See: [`WC_Order::get_order_number()`](https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-order.php#L441)